docs/adr/0039-locales-languages.md

```diff
@@ -0,0 +1,183 @@
+
+Parmi ces identifiants de locales spécifiques et historiques il y a les identifiants des *POSIX locales*, de la forme suivante :
+
+`[language[_territory][.codeset][@modifier]]`.
+
+Les identifiants de locales *POSIX locales* sont donc par exemple : `fr`, `fr_FR`, `fr_BE`, `fr_FR.UTF-8`, `en`, `en_GB`, `en_US`, et même `fr_FR@euro` pour un identifiant ancien datant du passage à l'euro, etc.
+
+Ces identifiants de locales sont toujours utilisés par les systèmes POSIX comme les distributions Linux, et aussi par certains outils existants depuis longtemps comme [GNU gettext utilities](https://www.gnu.org/software/gettext/manual/html_node/Locale-Environment-Variables.html). La plateforme Java aussi (construit suivant la tradition UNIX, car créé par Sun Microsystems, entreprise qui concevait et vendait des machines de type UNIX) se basait sur ces identifiants, gérés au niveau de la classe `java.util.Locale`.
+C'est pourquoi beaucoup peuvent se souvenir de l'utilisation de fichiers PO de la forme `fr_FR.po`, `en_US.po`, etc. et d'applications (notamment des applications Java multiplateformes) lancées en spécifiant `LC_ALL=fr_FR`, `LC_ALL=en_US`, etc.
+
+Mais avec la généralisation d'Unicode sur quasiment tous les systèmes et toutes les piles logicielles il y a eu l'opportunité et la volonté de standardiser pour n'avoir qu'un seul format d'identifiants de locales.
+
+La standardisation du format des identifiants de locales s'est faite dans la spécification [Tags for Identifying Languages (BCP 47 - RFC 5646)](https://www.rfc-editor.org/rfc/rfc5646.html), notamment autour de la notion de *language tags* définis . Il aurait peut-être été plus clair que ce mécanisme soit nommé *locale tags* … mais c'est le nom *language tags* qui a été retenu pour ce concept. Aussi pour éviter les confusions, dans le reste de ce document on va parler du *format BCP 47*.
+
+Concrètement avant de donner plus d'informations, voici des exemples d'identifiants de locales au *format BCP 47* :
+
+* `fr`
+* `fr-FR`
+* `fr-BE`
+* `en`
+* `en-US`
+* `en-GB`
+
+Le W3C définit le *format BCP 47* et les locales Unicode comme étant le format à utiliser par le web pour les identifiants de locales dans le document [Language Tags and Locale Identifiers for the World Wide Web](https://www.w3.org/TR/ltli/).
+
+Idem pour le langage JavaScript dans le namespace [`Intl`](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Intl) avec notamment le builtin `Intl.Locale`.
+
+La plateforme Java, dans sa classe [`java.util.Locale`](https://docs.oracle.com/en/java/javase/19/docs/api/java.base/java/util/Locale.html) définit également le *format BCP 47* comme étant le format à utiliser pour les identifiants de locales, mais continue d'utiliser la forme POSIX pour sa méthode `toString` uniquement pour des raisons de compatibilité tout en dépréciant cette utilisation depuis *Java 19*.
```

Comment:
Sur la première partie de l'historique, je trouve intéressant de rappeler l'utilisation faite du format par Java. 
En fin d'historique, quand on commence à revenir à notre contexte (full JS), je trouve que cela peut perdre un peu durant la lecture sur notre contexte.

Reply:
Je ressentais la même chose mais j'ai eu du mal à trouver une rédaction adéquate. Je viens d'ajouter un commit _« chore(docs): Group historical context at the beginning and group JavaScript context at the end »_ qui j'espère rend la progression de la lecture plus logique et agréable. Toute autre suggestion bienvenue !